urllib3/connectionpool.py

```diff
@@ -239,11 +239,15 @@ def _get_conn(self, timeout=None):
         # If this is a persistent connection, check if it got disconnected
         if conn and is_connection_dropped(conn):
             log.info("Resetting dropped connection: %s" % self.host)
-            conn.close()
-            if getattr(conn, 'auto_open', 1) == 0:
-                # This is a proxied connection that has been mutated by
-                # httplib._tunnel() and cannot be reused (since it would
-                # attempt to bypass the proxy)
+            try:
+                conn.close()
+                if getattr(conn, 'auto_open', 1) == 0:
+                    # This is a proxied connection that has been mutated by
+                    # httplib._tunnel() and cannot be reused (since it would
+                    # attempt to bypass the proxy)
+                    conn = None
+            except Exception as e:
```

Comment:
I disagree that it's a good idea to catch all errors and silently discard even in this case. We could have a syntax error in conn.close somehow, or a new scenario that we'd want to be aware of, or any number of conditions where this behaviour will hide what's going on underneath and make it a lot harder to debug.

We'll need to track down the specific exception and maybe add a wrapper that we can still catch even if pyopenssl is not available.